### ISSUE TITLE
Add no-dupe-class-members rule that supports ClassProperty

### ISSUE THREAD

**Issue by Turbo87** [babel/eslint-plugin-babel#190]
see https://github.com/eslint/eslint/issues/13059

```js
class A { foo = 42; get foo() {} }
```

If the rule will not be adjusted in the ESLint repo we should consider copying the rule here and adjusting it to work for Stage 3 class fields too.

**Comment by ljharb** [comment]
This is a strongly recommended pattern:
```js
class A {
  foo = this.foo.bind(this);
  foo() { … }
}
```
so the rule should *not* warn on those under any circumstances.

**Comment by Turbo87** [comment]
hmm, that seems like a somewhat strange pattern to me, but I guess it could be made configurable?

**Comment by ljharb** [comment]
It's the only way to have a bound own method, without using arrow functions (arrow functions should never be used in class fields), and preserving both borrowability and testability of the prototype methods.

Configuration is fine, but these must be allowed by default.

**Comment by Turbo87** [comment]
I just saw that there is actually already a PR for this (https://github.com/babel/eslint-plugin-babel/pull/173), though without such a config option.

@Graham42 would you have time to update the PR to add such a config option?

**Comment by Graham42** [comment]
I'm no longer using this library. If someone else would like to pick that PR, feel free to do so.

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [@babel/eslint-plugin has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
